File: 014_longest_common_prefix.cpp
#include <bits/stdc++.h>
using namespace std;
/**
 * * Question Link: https://leetcode.com/problems/longest-common-prefix/
 * * Sample Input:  strs = ["flower","flow","flight"]
 * * Sample Output: Output: "fl"
 */

class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        int shortest_length = INT32_MAX;

        for (auto string : strs)
        {
            shortest_length = min(shortest_length, (int)string.length());
        }

        string ans = "";
        for (int i = 0; i < shortest_length; i++)
        {
            char current = strs[0][i];
            for (int j = 1; j < strs.size(); j++)
            {
                if (strs[j][i] != current)
                {
                    return ans;
                }
            }
            ans += current;
        }
        return ans;
    }
};

File: 014_longest_common_prefix.ts
function longestCommonPrefix(strs: string[]): string {
    let ans = "";
    const shortestLength = strs.reduce(
        (acc, curr) => Math.min(acc, curr.length),
        Infinity
    );
    for (let i = 0; i < shortestLength; i++) {
        let isCommon = true;
        for (let currIndex = 1; currIndex < strs.length; currIndex++) {
            if (strs[currIndex][i] !== strs[currIndex - 1][i]) {
                return ans;
            }
        }
        ans += strs[0][i];
    }
    return ans;
}
